pkg/bindings/images/build.go

```diff
@@ -0,0 +1,148 @@
+package images
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"regexp"
+	"strconv"
+
+	"github.com/containers/buildah"
+	"github.com/containers/podman/v2/pkg/bindings"
+	"github.com/containers/podman/v2/pkg/domain/entities"
+	"github.com/docker/go-units"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/pkg/errors"
+)
+
+// Build creates an image using a containerfile reference
+func Build(ctx context.Context, containerFiles []string, tarfile io.Reader, options entities.BuildOptions) (*entities.BuildReport, error) {
+	conn, err := bindings.GetClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+	params := url.Values{}
+	params.Set("dockerfile", containerFiles[0])
```

Comment:
Not for  this review, just for rumination for later, could we change this to containerfile?

Reply:
@TomSweeneyRedHat the parameter name here is part the API, changing it would break compatibility.